### DISCUSSION TITLE
Selection of comments in facebook doesn't see them all.

### DISCUSSION THREAD

**Discussion by LightTemplar** [coddingtonbear/obsidian-web#151] [Bug Report]
I tried to add comments along with the post from FaceBook, but `page.selectedText` has found only one comment from all selected.
You can test it with this post, for example.
https://www.facebook.com/groups/655314275791511/posts/934448084544794/

**Comment by coddingtonbear** [comment]
Interesting!  I've never seen such a bug, but it does look like that's a real thing.

The logic we're using for extracting that text is here: https://github.com/coddingtonbear/obsidian-web/blob/f2ad5b45cc54d6407c10c3298e17c8eb8d97b1a9/src/popup.tsx#L220 -- if you can find the problem and correct it, I'd be glad to look at your pull request.

**Comment by LightTemplar** [comment]
I've experimented a bit in developer console, and found out, that `window.getSelection()` doesn't work correctly in this case. If there is a workaround - I don't know yet.

**Comment by coddingtonbear** [comment]
Unfortunately the `window.getSelection()` option would be a little difficult to accept even if it were working because that mechanism strips all formatting.  The weird thing I'm doing above is entirely to try to preserve as much of that as possible so you can copy also links, embedded image references, etc.
